snippets/serverless-kafka-broker-intro.adoc

```diff
@@ -1,7 +1,17 @@
 // Text snippet included in the following modules:
 //
 // * /serverless/develop/serverless-using-brokers.adoc
+// * /serverless/develop/serverless-kafka-developer.adoc
 
 :_content-type: SNIPPET
 
+The Knative Kafka broker is an Apache Kafka native implementation of the Knative broker API. It has a better integration with Kafka for the broker and trigger model over other broker types and reduces network hops.
```

Comment:
I think I'll leave it as is for now, since this information came from upstream and the main goal of the PR was really just to add the info about the binary content mode, so I don't want to get too sidetracked 🙂 Would be a good task for us later though in general to flesh out some of the Kafka docs I think, but there's a lot of flux with Knative Kafka in general atm anyway.